### PR TITLE
csharp sdk simplifier

### PR DESCRIPTION
**Pull request by VeryEarly**

### CODE REVIEW

File: powershell/models/TrieNode.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  isEnd: boolean;
+  constructor(value: string) {
+    this.value = value;
+    this.children = new Map<string, TrieNode>();
+    this.isEnd = false;
+  }
+
+  addChild(child: TrieNode): void
+  addChild(child: string): void
+  addChild(child: unknown): void {
```
Comment (dolauli) [inline powershell/models/TrieNode.ts:13]: If addChild only accepts TrieNode or String as input, suggest to change it to addChild(child: TrieNode | string).

File: powershell/models/TrieNode.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    } else if (child instanceof TrieNode && !this.children.has(child.value)) {
+      this.children.set(child.value, child);
+    } else {
+      throw new Error('invalid trie node or trie node value.');
+    }
+  }
+
+  hasChild(child: TrieNode): boolean
+  hasChild(child: string): boolean
+  hasChild(child: unknown): boolean {
```
Comment (dolauli) [inline powershell/models/TrieNode.ts:25]: Same as above.

File: powershell/models/TrieNode.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+export class TrieNode {
```
Comment (dolauli) [inline powershell/models/TrieNode.ts:1]: Please add license header for this file and move it to utils folder.

**Review by dolauli — CHANGES_REQUESTED** [review]
Looks good to me but some minor changes are needed.

File: powershell/models/TrieNode.ts (hunk 1 above)
Comment (VeryEarly) [reply powershell/models/TrieNode.ts]: fixed

File: powershell/models/TrieNode.ts (hunk 3 above)
Comment (VeryEarly) [reply powershell/models/TrieNode.ts]: fixed

File: powershell/models/TrieNode.ts (hunk 2 above)
Comment (VeryEarly) [reply powershell/models/TrieNode.ts]: fixed

**Review by dolauli — APPROVED** [review]
LGTM

**Pull request merged**